### PR TITLE
Fix loading/editing users that do not have a role assignment

### PR DESCRIPTION
**Pull request by JammingBen**
## Related Issue

- Fixes https://github.com/owncloud/web/issues/8585

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

### CODE REVIEW

**Comment by JanAckermann** [comment]
How does this happen, in which case a user doesn't have a role? This is quite new to me.

**Comment by JammingBen** [comment]
You can test this by running oCIS locally and clearing the `metadata` folder on your storage. I can't give you a real-world example though, maybe @micbar knows more about that.

But as the field can be empty and the API allows that, I guess we shouldn't assume that it's always set.

**Comment by JanAckermann** [comment]
image

image

We might want to show a '-' in the details and also in the table

File: packages/web-app-admin-settings/src/views/Users.vue [hunk 1]
```diff
@@ -693,7 +696,7 @@ export default defineComponent({
     updateUserAppRoleAssignments(user, editUser) {
       return this.graphClient.users.createUserAppRoleAssignment(user.id, {
         appRoleId: editUser.appRoleAssignments[0].appRoleId,
-        resourceId: editUser.appRoleAssignments[0].resourceId,
+        resourceId: editUser.appRoleAssignments[0].resourceId || this.applicationId,
```
Comment (JanAckermann) [inline packages/web-app-admin-settings/src/views/Users.vue:699]: could this always be this.applicationId to avoid confusion ?

**Review by JanAckermann — CHANGES_REQUESTED** [review]

**Comment by JammingBen** [comment]
> We might want to show a '-' in the details and also in the table

Agreed, also keeps it compatible with the Quota field 👍

File: packages/web-app-admin-settings/src/views/Users.vue (hunk 1 above)
Comment (JammingBen) [reply packages/web-app-admin-settings/src/views/Users.vue:699]: Should work that way, yes!

File: packages/web-app-admin-settings/src/components/Users/UsersList.vue [hunk 2]
```diff
@@ -346,8 +346,10 @@ export default defineComponent({
     getRoleDisplayNameByUser(user) {
       const assignedRole = user.appRoleAssignments[0]
 
-      return this.$gettext(
-        this.roles.find((role) => role.id === assignedRole?.appRoleId)?.displayName || ''
+      return (
+        this.$gettext(
+          this.roles.find((role) => role.id === assignedRole?.appRoleId)?.displayName || ''
+        ) || '-'
```
Comment (JanAckermann) [inline packages/web-app-admin-settings/src/components/Users/UsersList.vue]: is the '' )  || '-' not irrelevant?

**Review by JanAckermann — CHANGES_REQUESTED** [review]

**Review by JanAckermann — APPROVED** [review]

**Pull request merged**